src/lib/secrets/wallets.ml

```diff
@@ -58,6 +58,12 @@ let generate_new t : Public_key.Compressed.t Deferred.t =
   Public_key.Compressed.Table.add_exn t.cache ~key:pk ~data:keypair ;
   pk
 
+let delete ({cache; _} as t : t) (pk : Public_key.Compressed.t) :
+    (unit, [`Not_found]) Deferred.Result.t =
+  Hashtbl.remove cache pk ;
+  Deferred.Or_error.try_with (fun () -> Unix.remove (get_path t pk))
+  |> Deferred.Result.map_error ~f:(fun _ -> `Not_found)
```

Comment:
Seems like conceivably there could be more errors than Not_found? This kinda conflates Not_found in the list of wallets and Not_found on disk/some error deleting. One could argue that not_found on disk actually a success..
Maybe we don't need to differentiate them, not sure

Reply:
I don't think we need to differentiate at the time being